_stbt/core.py

```diff
@@ -1942,36 +1942,43 @@ def set_source_pipeline_playing(self):
 
         self.source_pipeline.set_state(Gst.State.PLAYING)
 
-    def _get_frame(self, timeout_secs=10):
-        try:
-            # Timeout in case no frames are received. This happens when the
-            # Hauppauge HDPVR video-capture device loses video.
-            gst_sample = self.last_sample.get(timeout=timeout_secs)
-        except Queue.Empty:
-            pipeline = self.source_pipeline
-            if pipeline:
-                Gst.debug_bin_to_dot_file_with_ts(
-                    pipeline, Gst.DebugGraphDetails.ALL, "NoVideo")
-            raise NoVideo("No video")
-        if isinstance(gst_sample, Exception):
-            raise UITestError(str(gst_sample))
-
-        if isinstance(gst_sample, Gst.Sample):
-            self.last_used_frame = array_from_sample(gst_sample)
-
-        return gst_sample
-
-    def pull_frame(self, timeout_secs=10):
-        return self._get_frame(timeout_secs)
+    def get_frame(self, timeout_secs=10, since=None):
+        import time
```

Comment:
How expensive is `import time`?

```
In [1]: import time

In [2]: %timeit import time
The slowest run took 43.52 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 367 ns per loop
```

Worst case: 367ns x 43.52 = 0.016ms = 1/1,000th of a frame at 60fps. That's the worst case (the slowest run). Maybe %timeit starts from a clean environment and imports it afresh?

Best case: 367ns = 1/45,000th of a frame at 60fps.